spec/fixtures/mauth_signature_testing.json

```diff
@@ -1,18 +1,109 @@
+      }
+    },
+    "single query": {
+      "query_string": "key1=a",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey1=a",
+        "signature": "PyWojze5RAo5c6oKUrBF1b861q1N0V2efXuZO0kVXTI5cWFbaYVIug5TALHy6r5yQD5yR0SvkXQgWaFO0vxn2ZtUcsVU9ZHMYVh+1aifW9XQMrRX44amYDZNozSUDhnUPncJXzuXy3IduL86EFAD0LkGTGsJK5nP9Do/dDihUDZXkjzifQVLWp7JNuS4Z06/ULEiY7gZGJDGGoLo0+2O/Zqf9um6rPvqan+u/IUD0R9LKBYBPPc3qcG9N555XizipAvzynrcFSJp3JmOKLpbM4ZqfPXVuJc1oiR5UgTEa2aPr5jUl3fdVk+cRRCH1IyUEu2ixecny3Q8jAE0ffij6g=="
+      }
+    },
+    "percent encoding": {
+      "unescaped_query": {
+        "key1": "あ",
+        "key2": "b"
+      },
+      "query_string": "key1=%E3%81%82&key2=b",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey1=%25E3%2581%2582&key2=b",
+        "signature": "kR19eT6Hk6gwl9ACxJUhzeQCVmcKdDMc1ZhNa5FHOh+CIgq03DNSvNh08kc7PNUiwYAC3QhkqLt0PPrysX+cf3Xertn3UQYN22+/S0JO4lkhgHv4s5HUKxtJSXJmAJssgGdQgFFnAdKJcJi3FH+02Cy1a4PMlWAjT9POyZQDS3cK7kvwwS8UyBrE8Cd1/DhBKsNyH0YQ8MsPArNXOGdFq2dH3HbuvszSTLBFj1piGVA3Sp1oPp8YlWKjmprw6HjMcGw7BLqLvFswq4zMd7QHzYyXehwScH7nMFwPPQccbZ6rh0kRR2o06me4QYu170Z0VHCna78UKqPzn8VJnAYFzg=="
+      }
+    },
+    "percent encoding (including lower case)": {
+      "unescaped_query": {
+        "key1": "あ",
+        "key2": "b"
+      },
+      "query_string": "key1=%e3%81%82&key2=b",
+      "v2":{
+        "string_to_sign": "PUT\n/v1/pictures\ncf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e\n5ff4257e-9c16-11e0-b048-0026bbfffe5e\n1309891855\nkey1=%25e3%2581%2582&key2=b",
+        "signature": "oZGM13Nx2/5JwD98pd52vUi3iEXTxj2BZBOp5Un+AZR3wKPfMk9PUKqXe4Vm4hdo5T148F742KdwG9mq6YsA7tbFETs8scv8Eqyha8nnXF8cyClT8X3YR0+z2jL9P8i/Pk39qHySd0xUKBQH93lG9dBTVjtKDLt8yK2QPmZYGZcXW7UAfuaXXSh9KBVUldOfwYJN1y+ceH1NmbIMAuRrHa/AS4y1NdZQNh08hR6YA5v+c6A5ttK93VEwOYal6iNchN+LhUuVHK8/iBg0rLQVEb0UIVwu8cEgcs6MRrskKcyZhc1rtHLLYVRLYMfdrwhYulp5omT7LqHALTsExZiOUA=="
+      }
+    },
+    "special characters": {
+      "unescaped_query": {
+        "key1": " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+      },
+      "query_string": "key1=+%21%22%23%24%25%26%27%28%29%2A%2B%2C-.%2F0123456789%3A%3B%3C%3D%3E%3F%40ABCDEFGHIJKLMNOPQRSTUVWXYZ%5B%5C%5D%5E_%60abcdefghijklmnopqrstuvwxyz%7B%7C%7D~",
```

Comment:
This query value also looks double-encoded. And "+" should be normalized to `%20`.

> For example, the space character must be encoded as %20 (not using '+', as some encoding schemes do) 
https://learn.mdsol.com/display/CA/MAuth%20Protocol%20V2%20Specification

Reply:
The application side generates this query string and passes to MAuth client.
In ruby, the space character will be encoded to `+` on the application side:
```ruby
CGI.escape " !\"#$%&'()*+,-./"
#=> "+%21%22%23%24%25%26%27%28%29%2A%2B%2C-.%2F"
```